Library/Homebrew/cmd/--cache.rb

```diff
@@ -29,11 +31,23 @@ def __cache
     if args.no_named?
       puts HOMEBREW_CACHE
     else
-      args.formulae.each do |f|
-        if Fetch.fetch_bottle?(f)
-          puts f.bottle.cached_download
+      args.named.each do |name|
+        formula = Formulary.factory name
+        if Fetch.fetch_bottle?(formula)
+          puts formula.bottle.cached_download
         else
-          puts f.cached_download
+          puts formula.cached_download
+        end
+      rescue FormulaUnavailableError => e
+        formula_error_message = e.message
+        begin
+          cask = Cask::CaskLoader.load name
+          puts "cask: #{Cask::Cmd::Cache.cached_location(cask)}"
```

Comment:
I think this command should only output the path. We should probably add `--formula` and `--cask` flags to allow specifying which path to show.